### PR TITLE
Update badge counts in new room list as needed

### PR DESCRIPTION
**Pull request by turt2live**
For https://github.com/vector-im/riot-web/issues/13635

Watch the 'Alt Grey Room' on the right side:
badges-real mp4

### CODE REVIEW

File: src/components/views/rooms/RoomTile2.tsx [hunk 1]
```diff
@@ -86,10 +88,19 @@ export default class RoomTile2 extends React.Component<IProps, IState> {
             hover: false,
             notificationState: this.getNotificationState(),
         };
+
+        // TODO: We shouldn't have to listen to every room update
+        // We don't have a model which works in a better way though.
+        MatrixClientPeg.get().on("Room.receipt", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Room.timeline", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Event.decrypted", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Room.redaction", this.tryUpdateIfRoomMatches);
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile2.tsx:97]: Any reason the 3 `Room.` event handlers are not bound on the room eventemitter itself to fire less?

Comment (turt2live) [reply src/components/views/rooms/RoomTile2.tsx:97]: Honestly just didn't expect it to work. If it does, then great! The TODO comment can go away with it.

File: src/components/views/rooms/RoomTile2.tsx [hunk 2]
```diff
@@ -89,17 +89,18 @@ export default class RoomTile2 extends React.Component<IProps, IState> {
...
 
-        // TODO: We shouldn't have to listen to every room update
-        // We don't have a model which works in a better way though.
-        MatrixClientPeg.get().on("Room.receipt", this.tryUpdateIfRoomMatches);
-        MatrixClientPeg.get().on("Room.timeline", this.tryUpdateIfRoomMatches);
-        MatrixClientPeg.get().on("Event.decrypted", this.tryUpdateIfRoomMatches);
-        MatrixClientPeg.get().on("Room.redaction", this.tryUpdateIfRoomMatches);
+        this.props.room.on("Room.receipt", this.handleRoomEventUpdate);
+        this.props.room.on("Room.timeline", this.handleRoomEventUpdate);
+        this.props.room.on("Event.decrypted", this.handleRoomEventUpdate);
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile2.tsx:94]: Doesn't look like this will get re-emit by the room sadly

Comment (t3chguy) [reply src/components/views/rooms/RoomTile2.tsx:94]: (the other 3 do and are fine)

**Pull request merged**